### PR TITLE
Update PyTensor dependency (drop numpy <2.0, and Python 3.10.)

### PR DESCRIPTION
**Pull request by ricardoV94**
Some major import changes in the latest release

----
📚 Documentation preview 📚: https://pymc--7910.org.readthedocs.build/en/7910/

### CODE REVIEW

**Comment by ricardoV94** [comment]
ugh loads of broken stuff

**Comment by ricardoV94** [comment]
Failures had to do with functions that guessed the inputs for the user. The order changed since the last release and now gradients / forward were not aligned.

This was not a great idea to begin with, so I now raise anytime such method is called in a way that the input order is ambiguous.

**Comment by ricardoV94** [comment]
CI broken by https://github.com/pymc-devs/pytensor/pull/1660 which means we'll need another release (which is also a major release as it drops old numpy / python) :)

**Comment by ricardoV94** [comment]
getting close. @maresb @lucianopaz any luck with the windows blas issue on conda?

**Comment by ricardoV94** [comment]
The verbose PR title is just to remind to add a note in the release notes

**Comment by ricardoV94** [comment]
I'm not testing on Windows because the release is held at conda-forge. I'll open an issue to revert the relevant commit once https://github.com/conda-forge/pytensor-suite-feedstock/pull/177 is fixed

**Comment by maresb** [comment]
Ya, sorry, I'm working on it, but not sure yet what the problem is. Will try to follow Luciano's advice

**Comment by maresb** [comment]
> Ya, sorry, I'm working on it, but not sure yet what the problem is. Will try to follow Luciano's advice

Solved! :tada:

**Comment by ricardoV94** [comment]
reverted the commit so we test on windows as usual

**Comment by ricardoV94** [comment]
Uhuh tests are passing

**Review by lucianopaz — APPROVED** [review]
I think the changes look OK, but maybe someone else should have a look too.

File: tests/sampling/test_jax.py [hunk 1]
```diff
@@ -45,6 +44,8 @@
     sample_numpyro_nuts,
 )
 
+MCMC = pytest.importorskip("numpyro.infer.MCMC")
```
Comment (Armavica) [inline tests/sampling/test_jax.py:47]: What is the behaviour of `importorskip` outside a test function? Does it skip all the rest of the file?

**Review by Armavica — APPROVED** [review]

File: tests/sampling/test_jax.py (hunk 1 above)
Comment (ricardoV94) [reply tests/sampling/test_jax.py:47]: The whole module

Comment (ricardoV94) [reply tests/sampling/test_jax.py:47]: Should change, good catch

Comment (jessegrabowski) [reply tests/sampling/test_jax.py:47]: This whole file depends on numpyro/blackjax, so we should be doing importorskip at the file level. Otherwise we have to do it on a test-by-test basis, which will be a lot more verbose

Comment (ricardoV94) [reply tests/sampling/test_jax.py:47]: okay then

**Comment by maresb** [comment]
Oops! We forgot to update python_requires:

https://github.com/pymc-devs/pymc/blob/f80e896a94f4fa6252e1a9991320de976a369109/setup.py#L73
